llama_index/response_synthesizers/refine.py

```diff
@@ -216,10 +217,26 @@ def _refine_response_single(
         logger.debug(f"> Refine context: {fmt_text_chunk}")
         if self._verbose:
             print(f"> Refine context: {fmt_text_chunk}")
+
         # NOTE: partial format refine template with query_str and existing_answer here
         refine_template = self._refine_template.partial_format(
             query_str=query_str, existing_answer=response
         )
+
+        # compute available chunk size to see if there is any available space
+        # determine if the refine template is too big (which can happen if
+        # prompt template + query + existing answer is too large)
+        avail_chunk_size = (
```

Comment:
Can we add this logic to `_arefine_response_single` too?

Reply:
yes, will do this!